src/mame/machine/k573dio.cpp

```diff
@@ -282,21 +282,21 @@ WRITE16_MEMBER(k573dio_device::ram_write_adr_low_w)
 
 READ16_MEMBER(k573dio_device::ram_r)
 {
-	uint16_t res = ram[(ram_read_adr & 0x7fffff) >> 1];
```

Comment:
I have no idea where 0x7fffff and 0x1ffffff came from in c910c1cf0ab357df250aff10002547e52e896f76.

These changes caused graphic corruptions (DDR 5th Mix), not being able to boot certain games (DDR Solo Bass Mix), hard locks (DDR Max/Max 2/Extreme/probably more), sound board errors (Drummania 9th Mix), among others.

@galibert If you have a specific reason for making these change then I would appreciate it if you could elaborate.

Reply:
7ffffff is a mistake, 1fffffff is I think correct.  If you don't mask and the game sends a too large value, you get an out-of-bounds error.

Reply:
Ok, I get what you were going for now. I will restore this change after some more testing. Thanks for the explanation.

Reply:
RAM address masks have been restored and tested with a few games (DDR 5th Mix, DDR Extreme, DM9). @galibert